apps/web/src/contexts/globalLayout/GlobalSidebar/SidebarDrawerContext.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  createContext,
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+
+export type DrawerType = 'Notifications' | 'Settings';
+
+type ActiveDrawer = {
+  content: ReactElement;
+  drawerType?: DrawerType;
+};
+
+type DrawerState = {
+  activeDrawer: ActiveDrawer | null;
+};
+
+const DrawerStateContext = createContext<DrawerState | undefined>(undefined);
+
+export const useDrawerState = (): DrawerState => {
+  const context = useContext(DrawerStateContext);
+  if (!context) {
+    throw new Error('Attempted to use DrawerStateContext without a provider!');
+  }
+
+  return context;
+};
+
+type DrawerActions = {
+  showDrawer: (props: ActiveDrawer) => void;
+  hideDrawer: () => void;
+};
+
+const DrawerActionsContext = createContext<DrawerActions | undefined>(undefined);
+
+export const useDrawerActions = (): DrawerActions => {
+  const context = useContext(DrawerActionsContext);
+  if (!context) {
+    throw new Error('Attempted to use DrawerActionsContext without a provider!');
+  }
+
+  return context;
+};
+
+type Props = { children: ReactNode };
+
+function isDrawerType(type: string): type is DrawerType {
+  console.log({ type });
```

Comment:
temporarily logging some stuff to debug why type guard returns false on preview deploy